### DISCUSSION TITLE
Naming convention

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#7422] [Legacy Topics]
Hello, What are the best practices when we name collection and fields of a collection?
Observation is that when we name a collection as APIDetails, its resource path or UID for the same would be api-details, which is automatically considered by the product.

If we do not have any best practices defined, then are there any limitations for naming the collection? Could it be anything? Only concern is that when we code queries, controller or model logic, we would not want to see any conflict with respect to naming.

**Comment by derrickmehaffy** [comment]
Naming should now be automatically handled when you create the model (in the admin panel) but one key thing to note right now as we have seen issues with it, make sure your model name has both a singular and a plural form.

The most recent example I have seen is `news` which in itself is both singular and plural and breaks when you try to use it. As for using the dash UIDs in queries and what not is no problem you just have to use something like `strapi.services['your-dash-model'].find()` instead of `strapi.services.yourmodel.find()`
